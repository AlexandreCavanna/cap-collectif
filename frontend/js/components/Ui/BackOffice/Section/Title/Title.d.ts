import { FC } from 'react'
import { TextProps } from '~ui/Primitives/Text'

declare const Title: FC<TextProps>

export default Title
