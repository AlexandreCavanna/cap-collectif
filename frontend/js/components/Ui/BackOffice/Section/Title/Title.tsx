import * as React from 'react'
import { headingStyles } from '~ui/Primitives/Heading'
import { FontWeight } from '~ui/Primitives/constants'
import type { TextProps } from '~ui/Primitives/Text'
import Text from '~ui/Primitives/Text'
type Props = TextProps & {
  readonly children: JSX.Element | JSX.Element[] | string
}

const Title = ({ children, ...rest }: Props) => (
  <Text {...headingStyles.h4} fontWeight={FontWeight.Semibold} color="blue.800" {...rest}>
    {children}
  </Text>
)

export default Title
