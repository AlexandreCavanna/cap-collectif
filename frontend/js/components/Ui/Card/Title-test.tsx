/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Title from './Title'

describe('<Title />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Title>My title</Title>)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly with other tagName', () => {
    const wrapper = shallow(<Title>My title</Title>)
    wrapper.setProps({
      tagName: 'h2',
    })
    expect(wrapper).toMatchSnapshot()
  })
})
