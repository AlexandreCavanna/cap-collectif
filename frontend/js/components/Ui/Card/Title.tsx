import * as React from 'react'
import type { StyledComponent } from 'styled-components'
import styled from 'styled-components'

const e = React.createElement
type ContainerProps = {
  tagName: string
  children: JSX.Element | JSX.Element[] | string
}
const Container: StyledComponent<any, {}, (arg0: ContainerProps) => JSX.Element | null | string> = styled(
  ({ tagName, children, ...props }) => e(tagName, props, children),
).attrs({
  className: 'card__title',
})`
  font-size: 18px;
  line-height: 1.2;
  margin: 0;
`

const Title = ({ tagName, children }: ContainerProps) =>
  tagName ? <Container tagName={tagName}>{children}</Container> : <>{children}</>

Title.defaultProps = {
  tagName: 'h3',
}
export default Title
