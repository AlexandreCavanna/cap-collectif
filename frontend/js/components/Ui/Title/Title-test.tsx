/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Title, { TYPE } from './Title'

describe('<Title />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Title type={TYPE.H1}>My title</Title>)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly with other type', () => {
    const wrapper = shallow(<Title type={TYPE.H2}>My title</Title>)
    expect(wrapper).toMatchSnapshot()
  })
})
