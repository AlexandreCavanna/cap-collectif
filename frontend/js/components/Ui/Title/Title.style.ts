import * as React from 'react'
import type { StyledComponent } from 'styled-components'
import styled from 'styled-components'
import type { TitleProps } from './Title'

const TitleContainer: StyledComponent<any, {}, (arg0: TitleProps) => React.ReactNode> = styled(
  ({ type, children, ...props }) => React.createElement(type, props, children),
).attrs({
  className: 'title',
})``
export default TitleContainer
