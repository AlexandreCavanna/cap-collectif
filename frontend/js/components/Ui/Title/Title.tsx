import { $Values } from 'utility-types'
import * as React from 'react'
import TitleContainer from './Title.style'

export const TYPE: {
  H1: 'h1'
  H2: 'h2'
  H3: 'h3'
} = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
}
export type TitleProps = {
  type: $Values<typeof TYPE>
  children: any
  className?: string
}

const Title = ({ type, className, children }: TitleProps) => (
  <TitleContainer type={type} className={className}>
    {children}
  </TitleContainer>
)

export default Title
