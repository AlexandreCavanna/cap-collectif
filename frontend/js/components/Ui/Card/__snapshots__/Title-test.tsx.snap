// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render correctly 1`] = `
<Title__Container
  tagName="h3"
>
  My title
</Title__Container>
`;

exports[`<Title /> should render correctly with other tagName 1`] = `
<Title__Container
  tagName="h2"
>
  My title
</Title__Container>
`;
